### PR TITLE
bugfix: fix configuration center can't read configuration using SpringCloudConfig

### LINKED ISSUE #2104: 配置中心使用SpringCloudConfig读取不到配置
- [x] I have searched the [issues](https://github.com/seata/seata/issues) of this repository and believe that this is not a duplicate.

### Ⅰ. Issue Description
seata1.0 使用springCloudConfig作为配置中心的时候，存在读取不到配置的情况（读不到就用了默认值）。 例如：

1. GlobalTransactionScanner去获取service.disableGlobalTransaction的配置的时候

2. GlobalTransactionScanner去获取datasource.autoproxy的配置的时候

经过调试发现，是在获取配置的时候，SpringContextProvider类还没有初始化好，其中的ApplicationContext和Environment为null。 

解决办法：
考虑让SpringContextProvider先初始化。之后自己打包了源码和修改，为SpringContextProvider实现了PriorityOrdered接口并实现getOrder方法返回highestorder。但是SpringContextProvide好像还是没有最先初始化，导致还是读取不到配置。

其他情况：
当SpringContextProvider类初始化后，就可以正常获取配置项，比如我的registry的配置（type:eureka... ）都是放在配置中心的，这些当程序跑到后面的时候又可以正常获取。估计就是SpringContextProvider初始化先后的问题？

### PR DESCRIPTION
**Pull request by xingfudeshi**
### Ⅰ. Describe what this PR did
Fix configuration center cannot read configuration using SpringCloudConfig

### Ⅱ. Does this pull request fix one issue?

fixes #2104 

### Ⅲ. Why don't you add test cases (unit test/integration test)? 
I've done integration testing and already added a UT.

### CODE REVIEW

File: config/seata-config-spring-cloud/src/main/java/io/seata/config/springcloud/SpringCloudConfiguration.java [hunk 1]
```diff
@@ -49,10 +50,10 @@ public String getTypeName() {
 
     @Override
     public String getConfig(String dataId, String defaultValue, long timeoutMills) {
-        if (null == SpringContextProvider.getEnvironment()) {
+        if (null == SeataSpringApplicationContextHolder.getApplicationContext() || null == SeataSpringApplicationContextHolder.getApplicationContext().getEnvironment()) {
```
Comment (jsbxyyx) [inline config/seata-config-spring-cloud/src/main/java/io/seata/config/springcloud/SpringCloudConfiguration.java]: extract `SeataSpringApplicationContextHolder.getApplicationContext()`

Comment (xingfudeshi) [reply config/seata-config-spring-cloud/src/main/java/io/seata/config/springcloud/SpringCloudConfiguration.java]: fixed

File: config/seata-config-spring-cloud/pom.xml [hunk 2]
```diff
@@ -18,8 +18,9 @@
             <version>${project.parent.version}</version>
         </dependency>
         <dependency>
-            <groupId>org.springframework</groupId>
-            <artifactId>spring-context</artifactId>
+            <groupId>io.seata</groupId>
+            <artifactId>seata-spring</artifactId>
+            <version>${project.parent.version}</version>
```
Comment (zjinlei) [inline config/seata-config-spring-cloud/pom.xml]: seata-spring depends on tm, rm, rm-datasource, etc. It is recommended to exclude all.

Comment (xingfudeshi) [reply config/seata-config-spring-cloud/pom.xml]: OK.

Comment (xingfudeshi) [reply config/seata-config-spring-cloud/pom.xml]: fixed.

**Pull request merged**